### PR TITLE
form validation

### PR DESCRIPTION
**Pull request by ChitvanRamani22**
https://user-images.githubusercontent.com/68965499/116811800-e8771980-ab68-11eb-981f-b595611f6a8d.mp4

@Namanl2001 @sagar-barapatre please review this pr.

### CODE REVIEW

**Comment by Namanl2001** [comment]
Hey, we can consider this PR but this does not solve the actual issue `#107`.
The issue `107` is for example if the user clicks the submit button without filling any detail in the form or let say missing an input which is mandatory to make a post request. You can check the model in the backend directory to check which are the mandatory inputs. And then we've to show the alerts at the missed input boxes which are mandatory to be filled out.  
Do you want to add the required change in a separate PR? as this PR looks good for its purpose. Thanks

File: src/components/update.jsx [hunk 1]
```diff
@@ -22,9 +22,42 @@ class update extends Component {
...
   };
 
   handleChange = e => {
+    let nam = e.target.id;
+    let val = e.target.value;
+    const pinRegex = RegExp(/[1-9][0-9]{5}/);
+    const phoneRegex = RegExp(/^[0-9\b]+$/);
+    let err = '';
+    if (nam === 'pin') {
+      if (!pinRegex.test(val) || (val.length != 6 && val !== '')) {
```
Comment (Namanl2001) [inline src/components/update.jsx:36]:
```suggestion
      if (!pinRegex.test(val) || (val.length !== 6 && val !== '')) {
```

File: src/components/update.jsx [hunk 2]
```diff
@@ -22,9 +22,42 @@ class update extends Component {
...
+        );
+        document.getElementById('pin').style.border = '1px solid red';
+      } else {
+        document.getElementById('pin').style.borderColor = '';
+      }
+      this.setState({ errormessage1: err });
+    }
+
+    if (nam === 'phone') {
+      if (!phoneRegex.test(val) || (val.length != 10 && val !== '')) {
```
Comment (Namanl2001) [inline src/components/update.jsx:48]:
```suggestion
      if (!phoneRegex.test(val) || (val.length !== 10 && val !== '')) {
```

File: src/components/form.jsx [hunk 3]
```diff
@@ -21,9 +21,42 @@ class form extends Component {
...
   };
 
   handleChange = e => {
+    let nam = e.target.id;
+    let val = e.target.value;
+    const pinRegex = RegExp(/[1-9][0-9]{5}/);
+    const phoneRegex = RegExp(/^[0-9\b]+$/);
+    let err = '';
+    if (nam === 'pin') {
+      if (!pinRegex.test(val) || (val.length != 6 && val !== '')) {
```
Comment (Namanl2001) [inline src/components/form.jsx:35]:
```suggestion
      if (!pinRegex.test(val) || (val.length !== 6 && val !== '')) {
```

File: src/components/form.jsx [hunk 4]
```diff
@@ -21,9 +21,42 @@ class form extends Component {
...
+        );
+        document.getElementById('pin').style.border = '1px solid red';
+      } else {
+        document.getElementById('pin').style.borderColor = '';
+      }
+      this.setState({ errormessage1: err });
+    }
+
+    if (nam === 'phone') {
+      if (!phoneRegex.test(val) || (val.length != 10 && val !== '')) {
```
Comment (Namanl2001) [inline src/components/form.jsx:47]:
```suggestion
      if (!phoneRegex.test(val) || (val.length !== 10 && val !== '')) {
```

**Review by Namanl2001 — CHANGES_REQUESTED** [review]
Small nits otherwise LGTM

**Comment by ChitvanRamani22** [comment]
> Hey, we can consider this PR but this does not solve the actual issue `#107`.
> The issue `107` is for example if the user clicks the submit button without filling any detail in the form or let say missing an input which is mandatory to make a post request. You can check the model in the backend directory to check which are the mandatory inputs. And then we've to show the alerts at the missed input boxes which are mandatory to be filled out.  
> Do you want to add the required change in a separate PR? as this PR looks good for its purpose. Thanks

@Namanl2001 i will create a seperate pr for that issue but for this pr should i create a seperate issue and attach this pr as a reference or you will directly merge it?

**Review by Namanl2001 — APPROVED** [review]
Thanks for the PR.

**Comment by Namanl2001** [comment]
>  will it get merged without any issue?

We already have an issue for that `#107`

**Comment by ChitvanRamani22** [comment]
> Thanks for the PR.

My pleasure!

**Pull request merged**